lib/resolvers.js

```diff
@@ -0,0 +1,97 @@
+class Resolvers {
+  constructor(context) {
+    this.Kuzzle = context.accessors.sdk;
+
+    this._resolversMap = {
+      Query: {
+        allBooks: async () => {
+          const results = await this.Kuzzle.document.search(
+            'test-data',
+            'books'
+          );
+
+          if (!results.hits.length) {
+            throw new Error('No books were found!');
```

Comment:
This should be a `NotFoundError`

Reply:
I would not trigger an error if  no document is found. Just zero results, since semantically, there's no error, just an information to be returned: there are no books in the shelf.
On the other hand, I'd wrap the whole thing into a `try...catch` block and reformat Kuzzle errors if necessary.